test/crossbrowser/src/browser/api.ts

```diff
@@ -0,0 +1,362 @@
+import {
+  ResourceID,
+  add,
+  lookup,
+  remove
+} from './resources';
+
+import {
+  serializeLocalParticipant,
+  serializeLocalTrack,
+  serializeLocalTrackPublication,
+  serializeRoom
+} from './serialize';
+
+declare const Twilio: any;
+
+/**
+ * Connect to a {@link Room}.
+ * @param {Array<*>} args
+ * @param {(room: Room) => void} sendRoomEvents
+ * @returns {Promise<object>}
+ */
+export async function connect(args: any, sendRoomEvents: (room: any) => void): Promise<any> {
+  const [ token, options ] = args;
+  let room: any;
+
+  try {
+    room = await Twilio.Video.connect(token, options);
+  } catch (e) {
+    return {
+      error: {
+        code: e.code,
+        message: e.message
+      }
+    };
+  }
+
+  add(room);
+  add(room.localParticipant);
+  room.localParticipant.tracks.forEach(add);
+  room.localParticipant.trackPublications.forEach(add);
```

Comment:
I'm not sure what `add` does yet, but could we let some LocalParticipant-specific version of `add` do the work of adding `localParticipant.tracks` and `localParticipant.trackPublications`? That way we aren't doing all this work in `connect`?